src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseDetailsEndpointPostStateIT.java

```diff
@@ -0,0 +1,127 @@
+        scripts = {"classpath:sql/insert_cases.sql"})
+    public void shouldChangeCaseStateWhenPostStateConditionMatches() throws Exception {
+        final JsonNode data = createData();
+        performRequestAndValidate(data, TEST_EVENT_ID, "state4");
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD,
+        scripts = {"classpath:sql/insert_cases.sql"})
+    public void shouldChangeCaseStateBasedOnPriorityWhenPostStateConditionMatches() throws Exception {
+        final JsonNode data = createData();
+        performRequestAndValidate(data, NO_PRE_STATES_EVENT, "CaseEnteredIntoLegacy");
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD,
+        scripts = {"classpath:sql/insert_cases.sql"})
+    public void shouldChangeToDefaultWhenPostStateConditionNotMatches() throws Exception {
+        final JsonNode data = mapper.readTree("{"
+            + "\"PersonLastName\":\"Test Last1\","
+            + "\"PersonFirstName\":\"Test First\"}");
+        performRequestAndValidate(data, NO_PRE_STATES_EVENT, "state4");
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD,
+        scripts = {"classpath:sql/insert_cases.sql"})
+    public void shouldStayInThePreviousStateWhenWildcardUsedInPostConditionState() throws Exception {
+        final JsonNode data = createData();
+        performRequestAndValidate(data, HAS_PRE_STATES_EVENT, "CaseCreated");
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD,
+        scripts = {"classpath:sql/insert_cases.sql"})
+    public void shouldIgnorePostStateConditionWhenFieldNotAssignedToEvent() throws Exception {
+        final JsonNode data = createData();
+        performRequestAndValidate(data, "Goodness", "CaseEnteredIntoLegacy");
+    }
+
+    private JsonNode createData() throws com.fasterxml.jackson.core.JsonProcessingException {
+        return mapper.readTree("{"
+            + "\"PersonLastName\":\"Test Last\","
+            + "\"PersonFirstName\":\"Test First\"}");
+    }
+
+    private void performRequestAndValidate(JsonNode data, String eventId, String expectedCaseState) throws Exception {
+        final String caseReference = "1601933818308168";
+        final String URL = "/citizens/" + UID + "/jurisdictions/" + JURISDICTION + "/case-types/"
+            + CASE_TYPE_POST_STATE + "/cases/" + caseReference + "/events";
+
+        final CaseDataContent caseDetailsToSave = newCaseDataContent().build();
+        final Event triggeringEvent = anEvent().build();
+        triggeringEvent.setEventId(eventId);
+        caseDetailsToSave.setEvent(triggeringEvent);
+        final String token = generateEventTokenNewCase(UID, JURISDICTION, CASE_TYPE_POST_STATE, eventId);
+        caseDetailsToSave.setToken(token);
+        caseDetailsToSave.setData(JacksonUtils.convertValue(data));
+
+        final MvcResult mvcResult = mockMvc.perform(post(URL)
```

Comment:
No need to store the result. mvcResult not used later in the code.

Reply:
removed..